distributed/core.py

```diff
@@ -280,7 +280,7 @@ def __await__(self):
         async def _():
             timeout = getattr(self, "death_timeout", 0)
             async with self._startup_lock:
-                if self.status in Status.ANY_RUNNING:
+                if self.status not in {Status.undefined, Status.created, Status.init}:
                     return self
```

Comment:
start does nothing if the worker is `Status.closing`
https://github.com/dask/distributed/blob/39c5e8856cb05ad1833a0f6ee6c3fea0ddab1c1c/distributed/worker.py#L1522-L1527
then `__await__` immediately sets the worker to `Status.running` this is definitely not correct

Reply:
Nanny has an interesting behaviour with `Nanny.__await__` and `Nanny.start()`: https://github.com/dask/distributed/blob/39c5e8856cb05ad1833a0f6ee6c3fea0ddab1c1c/distributed/nanny.py#L341-L345

it sets its own `self.status = Status.running` or calls `self.close()`  - but if you `await Nanny(...)`  then whatever `Nanny.start()` does to `self.status` gets set back to `self.status = Status.running`

Reply:
this also interacts badly with `SpecCluster._correct_state_internal` https://github.com/dask/distributed/blob/39c5e8856cb05ad1833a0f6ee6c3fea0ddab1c1c/distributed/deploy/spec.py#L348-L352

`asyncio.wait(` doesn't propagate exceptions and so a worker can fail to start, and then be awaited a second time in `await w  # for tornado gen.coroutine support `